include/xtensor/xadapt.hpp

```diff
@@ -100,7 +100,7 @@ namespace xt
      */
     template <class C, std::size_t N>
     xtensor_adaptor<C, N, layout_type::dynamic>
-    xadapt(C& container, const std::array<typename C::size_type, N>& shape, const std::array<typename C::size_type, N>& strides);
+    xadapt(C& container, const stat_shape<typename C::size_type, N>& shape, const stat_shape<typename C::size_type, N>& strides);
```

Comment:
I agree that `xshape<3>` and `xshape<>` are better alias types in the long run. During the experimentation phase, the present definition has the advantage of making the old and new shape types exchangeable.